File: app/src/main/java/com/leoart/koreanphrasebook/data/network/firebase/dialogs/models/DialogParser.java
package com.leoart.koreanphrasebook.data.network.firebase.dialogs.models;

import java.io.BufferedReader;
import java.io.FileInputStream;
import java.io.IOException;
import java.io.InputStreamReader;

/**
 * @author Bogdan Ustyak (bogdan.ustyak@gmail.com)
 */

public class DialogParser {
    private final String filePath;

    public DialogParser(String filePath) {
        this.filePath = filePath;
    }

    public Dialog parse() throws IOException {

        Dialog dialog = new Dialog();

        // Open the file
        FileInputStream fstream = new FileInputStream(filePath);
        BufferedReader br = new BufferedReader(new InputStreamReader(fstream));

        int counter = 0;
        int number = 0;
        String ukrainian;
        String korean = "";
        while ((ukrainian = br.readLine()) != null && (korean = br.readLine()) != null) {
            number++;
            counter++;
            dialog.addMessage(new Replic(korean, ukrainian, number));
        }

        //Close the input stream
        br.close();

        return dialog;
    }
}

File: app/src/main/java/com/leoart/koreanphrasebook/data/parsers/dialogs/DialogParser.kt
package com.leoart.koreanphrasebook.data.parsers.dialogs

import android.content.Context
import com.leoart.koreanphrasebook.data.network.firebase.DialogsRequest
import java.io.IOException

/**
 * @author Bogdan Ustyak (bogdan.ustyak@gmail.com)
 */

class DialogParser(private val context: Context, private val fileName: String) {

    @Throws(IOException::class)
    fun parse() {

        val dialog = DialogsStream(context.assets.open(fileName)).parse()
        DialogsRequest().writeDialogReplics(dialog)
    }
}
